src/remark/code-block-snippets.js

```diff
@@ -24,8 +24,16 @@ let mdFilePath = "";
  * @typedef {import('mdast').Code} CodeNode
  * @param {Readonly<Omit<ImageNode, 'type'>> & CodeNode} node
  */
+
+/**
+ * Find paragraph -> image pattern and perform the replacement.
+ */
 function visitor(node) {
-    if (!node.alt?.startsWith("code")) return;
+    if (!node.children || node.children[0]?.type != 'image' || !node.children[0]?.alt?.startsWith("code")) return;
+
+    // Move code block in place of the paragraph node.
+    Object.assign(node, node.children[0]);
+    delete node.children;
```

Comment:
I'm not sure what issue this PR is supposed to solve, but this can eat up content:
```md
# Getting Started

![code js](../src/remark/code-block-snippets.test.js#test-region)
hello

world

## Use Oasis
```

| Before | After |
| --- | --- |
| ![Screenshot from 2022-09-27 21-37-56](https://user-images.githubusercontent.com/3758846/192623684-39de4959-15b3-40b3-a4d3-4c8e40bd6e3e.png)  | ![Screenshot from 2022-09-27 21-34-57](https://user-images.githubusercontent.com/3758846/192623679-27a29cac-5d1a-4039-b179-6000b4d66a50.png)  |

Reply:
Good catch. I made a fix to require only a single child containing the image node. Otherwise it ignores the external code snippet.

Reply:
Perhaps it would be even better to throw an error in that case so developers notice

Reply:
Ah, it does give a warning
```
[WARNING] The image at "..../src/remark/code-block-snippets.test.js" can't be read correctly. Please ensure it's a valid image.
unsupported file type: undefined (file: ..../src/remark/code-block-snippets.test.js)
```